src/com.company/heap/ProductOfThree.java

```diff
@@ -0,0 +1,104 @@
+package heap;
+
+import java.util.Arrays;
+import java.util.PriorityQueue;
+
+public class ProductOfThree {
+    private int[] findProductOfThree(int[] arr) {
```

Comment:
This is the wrong logic. This problem needs to be reworked. We shall discuss in the call about this issue.